File: bench/bintree.py
class Node:
    def __init__(self, d):
        self.l = None
        self.r = None
        if d > 0:
            self.l = Node(d-1)
            self.r = Node(d-1)

def check(n):
    return check(n.l) + check(n.r) + 1 if n.l else 1

n = 21
s = 0
for _ in range(10):
    t = Node(n)
    s += check(t)
print(s)

File: bench/bintree.php
<?php
class Node { public $l, $r; }
function make($d) {
    $n = new Node;
    if ($d > 0) { $n->l = make($d-1); $n->r = make($d-1); }
    return $n;
}
function check($n) {
    return $n->l ? check($n->l) + check($n->r) + 1 : 1;
}
$n = 21; $s = 0;
for ($i = 0; $i < 10; $i++) {
    $t = make($n);
    $s += check($t);
}
echo "$s\n";
